spring-cloud-deployer-kubernetes/src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesTaskLauncher.java

```diff
@@ -221,9 +221,16 @@ public String getLog(String id) {
 	}
 
 	private boolean maxConcurrentExecutionsReached() {
+		if (isMaximumConcurrentTasksUnlimited()) {
+			return false;
+		}
 		return this.getRunningTaskExecutionCount() >= this.getMaximumConcurrentTasks();
```

Comment:
How about `return this.getMaximumConcurrentTasks() > 0 && this.getCurrentTaskExecutionCount() >= this.getMaximumConcurrentTasks();`
This would allow a value of 0 to indicate no limit since 0 would normally mean no tasks can be executed